packages/replace/src/index.js

```diff
@@ -47,39 +47,49 @@ export default function replace(options = {}) {
     ? new RegExp(`${escape(delimiters[0])}(${keys.join('|')})${escape(delimiters[1])}`, 'g')
     : new RegExp(`\\b(${keys.join('|')})\\b`, 'g');
 
-  return {
-    name: 'replace',
+  function executeReplacement(code, id) {
+    const magicString = new MagicString(code);
 
-    transform(code, id) {
-      if (!keys.length) return null;
-      if (!filter(id)) return null;
+    let hasReplacements = false;
+    let match;
+    let start;
+    let end;
+    let replacement;
 
-      const magicString = new MagicString(code);
+    // eslint-disable-next-line no-cond-assign
+    while ((match = pattern.exec(code))) {
```

Comment:
seems like it'll be cleaner if we pull this assignment to L54

Reply:
I can change this up if you'd prefer that to be restructured. My goal was to be as minimally intrusive as possible, so this is just the contents of the original `transform` function.

Reply:
changes made!